pkg/db/queries.sql

```diff
@@ -75,3 +75,46 @@ FROM
 GROUP BY
 	originator_node_id;
 
+-- name: GetAddressLogs :many
+SELECT
+	a.address,
+	encode(a.inbox_id, 'hex') AS inbox_id,
+	a.association_sequence_id
+FROM
+	address_log a
+	INNER JOIN (
+		SELECT
+			address,
+			MAX(association_sequence_id) AS max_association_sequence_id
+		FROM
+			address_log
+		WHERE
+			address = ANY (@addresses::TEXT[])
+			AND revocation_sequence_id IS NULL
+		GROUP BY
+			address) b ON a.address = b.address
+	AND a.association_sequence_id = b.max_association_sequence_id;
+
+-- name: InsertAddressLog :exec
```

Comment:
@neekolas - Very nitpicky, but for idempotent inserts I think it's always better to use `:execrows` if we're not using `:one`, even if we ignore the result in go code via `_, err = `, because it makes the intent explicit to readers (we want to do nothing if the row already exists).

In this case though, because we have a primary key on `address, inbox_id` (a change from the old code), it means that an address is being added to an inbox_id but a row already exists for the `(address, inbox_id)` pair -
- If the row is revoked with a lower sequence ID, we should replace the row
- If the row is unrevoked and has a higher sequence ID, we should either surface an error or silently return success

Could probably change the query to do something like `... ON CONFLICT WHERE revocation_sequence_id IS NULL OR revocation_sequence_id < @association_sequence_id AND association_sequence_id < @association-sequence_id DO UPDATE ...`

Reply:
This is a very good point. In the fog of last week's hackathon I forgot that we changed the schema, so I left the query pretty much intact. Will push up a change to update the query